hbase-protocol-shaded/src/main/protobuf/AccessControl.proto

```diff
@@ -128,6 +128,15 @@ message HasUserPermissionsResponse {
     repeated bool has_user_permission = 1;
 }
 
+message HasPermissionRequest {
+  required TablePermission table_permission = 1;
+  required bytes user_name = 2;
+}
```

Comment:
Missing from hbase-protocol-shaded?

Reply:
Yes. This is only used in CPEP in the past.